### ISSUE TITLE
Setting `key = "boolean"` in schema and while inserting `key = false` is not working

### ISSUE THREAD

**Issue by EMPAT94** [tami5/sqlite.lua#123]
**Minimal example :**

```lua
local sqlite = require("sqlite")

local db = sqlite {
  uri = "test_db",
  tbl = {
    ok_key = "text",
    problem_key = "boolean",
  },
}

function main()
  if (db:close()) then db:open() end

  db:insert("tbl", {
    ok_key = "this works",
    problem_key = false,
    -- this is either silently ignored and not inserted in db,
    -- or throws "Missing Key" error when set to false
  })
  db:close()
end

main()
```

**Schema :**
```sh
sqlite-test · sqlite3 test_db
SQLite version 3.36.0 2021-06-18 18:36:39
Enter ".help" for usage hints.
sqlite> .tables
tbl
sqlite> .schema tbl
CREATE TABLE tbl(ok_key text, problem_key boolean);
```

**Trace :**
```sh
luajit: /usr/local/share/lua/5.1/sqlite/assert.lua:62: sqlite.lua: (insert) missing a required key: table: 0x416e42c0
stack traceback:
	[C]: in function 'assert'
	/usr/local/share/lua/5.1/sqlite/assert.lua:62: in function 'missing_req_key'
	/usr/local/share/lua/5.1/sqlite/parser.lua:537: in function 'f'
	/usr/local/share/lua/5.1/sqlite/utils.lua:154: in function 'map'
	/usr/local/share/lua/5.1/sqlite/parser.lua:534: in function 'pre_insert'
	/usr/local/share/lua/5.1/sqlite/db.lua:443: in function 'insert'
	test.lua:14: in function 'main'
	test.lua:23: in main chunk
	[C]: at 0x55f144900c20
```

**Environment :**

- neovim v0.5 (Stable)
- luajit v2.0.5
- sqlite3 v3.36
- sqlite v1.2.0-0 

_Thanks for the aweome plugin btw! Please let me know if there is anything else I can provide to reproduce the issue._

**Comment by tami5** [comment]
hmmm I think I see what's happening, false is treated as nil. Could please you confirm that inserting value true works.

Thanks

**Comment by EMPAT94** [comment]
yep, true works fine.

**Comment by tami5** [comment]
Nice this should be an easy fix in how values get converted works.

I'll push a fix in a pit, Thanks for debugging.

**Comment by tami5** [comment]
@EMPAT94 I believe is fixed now, can you confirm.

**Comment by EMPAT94** [comment]
I'm using sqlite installed via luarocks, not as a plug. Any pointers on how I should test the changes you've made ?

**Comment by tami5** [comment]
@EMPAT94 You can install master branch using luarocks till the next release.

```bash
luarocks install sqlite master
```

Not sure if you have to remove current one, but is good to do `luarocks remove sqlite` first

**Comment by EMPAT94** [comment]
TL;DR = I'm afraid I was not able to test the commit satisfactorily. Installing `sqlite master` didn't seem to change the result, but it may be an issue unrelated to sqlite itself. When I manually changed the file where error was occuring, it worked just fine.

 These are the steps I followed -

**test.lua same as before**

**1. Remove sqlite 1.2 and install 1.42 (master)**

```sh
/tmp · sudo luarocks remove sqlite && sudo luarocks install sqlite master
Checking stability of dependencies in the absence of
sqlite v1.2.0-0...

Removing sqlite v1.2.0-0...
Removal successful.
Installing https://luarocks.org/sqlite-master-0.src.rock

sqlite master-0 depends on luv (1.42.0-0 installed)
sqlite master-0 is now installed in /usr/local (license: MIT)
```

**2. Try running test.lua**

```sh
/tmp · lua test.lua
luajit: /usr/local/share/lua/5.1/sqlite/assert.lua:28: sqlite.lua: can not execute insert, tbl doesn't
exists.
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/sqlite/assert.lua:28: in function 'is_sqltbl'
        /usr/local/share/lua/5.1/sqlite/db.lua:411: in function 'insert'
        test.lua:14: in function 'main'
        test.lua:23: in main chunk
        [C]: at 0x55c17cae2c20
```

**3. Looks like there was some change in how database/tables are created. So I reinstalled 1.2 (stable) sqlite and made sure the table exists.**

```sh
/tmp · sudo luarocks remove sqlite && sudo luarocks install sqlite
Checking stability of dependencies in the absence of
sqlite master-0...

Removing sqlite master-0...
Removal successful.
Installing https://luarocks.org/sqlite-v1.2.0-0.src.rock

sqlite v1.2.0-0 depends on luv (1.42.0-0 installed)
sqlite v1.2.0-0 is now installed in /usr/local (license: MIT)

/tmp · lua test.lua
luajit: /usr/local/share/lua/5.1/sqlite/assert.lua:62: sqlite.lua: (insert) missing a required key: tab
le: 0x41997a70
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/sqlite/assert.lua:62: in function 'missing_req_key'
        /usr/local/share/lua/5.1/sqlite/parser.lua:537: in function 'f'
        /usr/local/share/lua/5.1/sqlite/utils.lua:154: in function 'map'
        /usr/local/share/lua/5.1/sqlite/parser.lua:534: in function 'pre_insert'
        /usr/local/share/lua/5.1/sqlite/db.lua:443: in function 'insert'
        test.lua:14: in function 'main'
        test.lua:23: in main chunk
        [C]: at 0x55a20e2b4c20
/tmp · sqlite3 test_db
SQLite version 3.36.0 2021-06-18 18:36:39
Enter ".help" for usage hints.
sqlite> .schema tbl
CREATE TABLE tbl(ok_key text, problem_key boolean);
sqlite> select * from tbl;
sqlite>
```

**4. Then I reinstalled 1.42 (master) sqlite and tried running test.lua.**

```sh
/tmp · sudo luarocks remove sqlite && sudo luarocks install sqlite master
Checking stability of dependencies in the absence of
sqlite v1.2.0-0...

Removing sqlite v1.2.0-0...
Removal successful.
Installing https://luarocks.org/sqlite-master-0.src.rock

sqlite master-0 depends on luv (1.42.0-0 installed)
sqlite master-0 is now installed in /usr/local (license: MIT)

/tmp · luarocks show sqlite

sqlite master-0 - SQLite/LuaJIT binding and a highly opinionated wrapper for storing, retrieving, cachi
ng, and persisting [SQLite] databases



License:        MIT
Homepage:       https://github.com/tami5/sqlite.lua
Labels:         sqlite3, binding, luajit, database
Installed in:   /usr/local

Modules:
        lua.sqlite.db.lua (/usr/local/share/lua/5.1/lua/sqlite/db/lua.lua)
        lua.sqlite.defs.lua (/usr/local/share/lua/5.1/lua/sqlite/defs/lua.lua)
        lua.sqlite.helpers.lua (/usr/local/share/lua/5.1/lua/sqlite/helpers/lua.lua)
        lua.sqlite.init.lua.init (/usr/local/share/lua/5.1/lua/sqlite/init/lua/init.lua)
        lua.sqlite.json.lua (/usr/local/share/lua/5.1/lua/sqlite/json/lua.lua)
        lua.sqlite.tbl.cache.lua (/usr/local/share/lua/5.1/lua/sqlite/tbl/cache/lua.lua)
        lua.sqlite.tbl.extend.lua (/usr/local/share/lua/5.1/lua/sqlite/tbl/extend/lua.lua)
        lua.sqlite.tbl.lua (/usr/local/share/lua/5.1/lua/sqlite/tbl/lua.lua)
        lua.sqlite.utils.lua (/usr/local/share/lua/5.1/lua/sqlite/utils/lua.lua)
        sql (/usr/local/share/lua/5.1/sql.lua)
        sqlite.assert (/usr/local/share/lua/5.1/sqlite/assert.lua)
        sqlite.db (/usr/local/share/lua/5.1/sqlite/db.lua)
        sqlite.defs (/usr/local/share/lua/5.1/sqlite/defs.lua)
        sqlite.helpers (/usr/local/share/lua/5.1/sqlite/helpers.lua)
        sqlite.init (/usr/local/share/lua/5.1/sqlite/init.lua)
        sqlite.json (/usr/local/share/lua/5.1/sqlite/json.lua)
        sqlite.parser (/usr/local/share/lua/5.1/sqlite/parser.lua)
        sqlite.stmt (/usr/local/share/lua/5.1/sqlite/stmt.lua)
        sqlite.strfun (/usr/local/share/lua/5.1/sqlite/strfun.lua)
        sqlite.tbl (/usr/local/share/lua/5.1/sqlite/tbl.lua)
        sqlite.tbl.cache (/usr/local/share/lua/5.1/sqlite/tbl/cache.lua)
        sqlite.tbl.extend (/usr/local/share/lua/5.1/sqlite/tbl/extend.lua)
        sqlite.utils (/usr/local/share/lua/5.1/sqlite/utils.lua)

Depends on:
        luv (using 1.42.0-0)

Indirectly pulling:
        luv (using 1.42.0-0)

/tmp · lua test.lua
luajit: /usr/local/share/lua/5.1/sqlite/assert.lua:62: sqlite.lua: (insert) missing a required key: tab
le: 0x404a7690
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/sqlite/assert.lua:62: in function 'missing_req_key'
        /usr/local/share/lua/5.1/sqlite/parser.lua:537: in function 'f'
        /usr/local/share/lua/5.1/sqlite/utils.lua:154: in function 'map'
        /usr/local/share/lua/5.1/sqlite/parser.lua:534: in function 'pre_insert'
        /usr/local/share/lua/5.1/sqlite/db.lua:414: in function 'insert'
        test.lua:14: in function 'main'
        test.lua:23: in main chunk
        [C]: at 0x55d43611ac20
```
**5. Just to be sure, I changed the condition in assert.lua:62 per your commit in my local file (added ~= nil condition)**

```sh
/tmp · lua test.lua
/tmp · sqlite3 test_db
SQLite version 3.36.0 2021-06-18 18:36:39
Enter ".help" for usage hints.
sqlite> select * from tbl;
this works|0
sqlite>
```

**_And it worked!_** Your commit is probably valid, but there was some complication in installing sqlite/master or my local setup, so I couldn't test it normally. 

What next steps do you suggest?

**Comment by tami5** [comment]
Glad we are over the fix doesn't work and their issue with installation method :D. I'm so noob at luarocks stuff, maybe luarocks master isn't what we expect it to be (i.e. build from master directly). I'd suggest installing sqlite using neovim until #126 is closed

**Comment by EMPAT94** [comment]
I'll try to avoid boolean for now and stay on stable. Thanks for your help so far!

**Comment by tami5** [comment]
> I'll try to avoid boolean for now and stay on stable. Thanks for your help so far!

We don't want that 😄 

I'll try to release v1.2.1 next week 🙂 so it can work according to your current setup.

**Comment by EMPAT94** [comment]
I appreciate that a lot @tami5. As for your confusion in me sticking to stable, perhaps I should explain better - I'm developing a new nvim plugin. So it is important that my dependencies are stable or i'll be chasing random bugs across huge codebases that I have no clue of. 🙅🏼 

PS: I'm a noobie myself, my first plugin and first lua project.  😄
